core/integration/module_config_test.go

```diff
@@ -1193,3 +1193,110 @@ func (m *Test) Fn(dir *dagger.Directory) *dagger.Directory {
 	require.NoError(t, err)
 	require.Equal(t, "", strings.TrimSpace(out))
 }
+
+func (ConfigSuite) TestDepPins(ctx context.Context, t *testctx.T) {
```

Comment:
It would be very nice to have a test where the pinned commit is behind HEAD on the branch and we verify that the pinned commit is used correctly

Reply:
Added a test for this using https://github.com/dagger/dagger-test-modules/pull/5, and another one for testing that `dagger develop` doesn't modify the pins.

Reply:
Approved that one, will let you press the merge button in case anything else needed